lib/pbench/server/indexer.py

```diff
@@ -1536,121 +1537,113 @@ def _make_source_json(self):
         We only process the top-level result.json file since it has all the
         data we need in one place.
         """
-        for dirname in self.json_dirs:
-            if dirname.startswith("sample"):
-                # sample result.json - we ignore each sample result.json since
-                # each sample is contained in the experiment level
-                # result.json.
-                continue
-            elif dirname != self.ptb.dirname:
-                # iteration result.json - there is no value in processing the
-                # result.json file at the iteration level because everything
-                # it contains is also in the experiment level's result.json.
-                continue
+        result_json = os.path.join(
```

Comment:
Consider using `Pathlib` here.

Reply:
I wonder if I should replace all the `os.path` with `pathlib.Path`?